internal/route/request_executor.go

```diff
@@ -34,18 +35,31 @@ func (e *HandledError) Error() string {
 }
 
 type OriginError struct {
-	err error
+	err          error
+	response     string
+	ResponseCode int
+}
+
+type HTTPResponse struct {
+	StatusCode int
```

Comment:
just curious, what was the reason behind using this type instead of `http.Response`?

Reply:
My intuition says that the response body, aka `HTTP.Response.Body`, should be read and discarded (`body.close()`) as soon as possible. Otherwise, the connection is held until `body.close()` is called.

And since `http.Response.body` is a field on `http.Response`,  it seems to appropriate to apply the "read and discard ASAP" model to the entire object. If we need some data from the object, it's best to read it out into an user-defined structure.

Besides that, it seems idiomatically weird to pass around an `http.Response` object due to the aforementioned reasons.